src/main/java/org/venice/beachfront/bfapi/services/JobService.java

```diff
@@ -50,8 +50,18 @@
 	 */
 	public Job createJob(String jobName, String creatorUserId, String sceneId, String algorithmId, String planetApiKey, Boolean computeMask,
 			JsonNode extras) throws UserException {
-		// Fetch the Algorithm and Scene information; Activate Scene if needed
+		// Fetch the Algorithm
 		Algorithm algorithm = algorithmService.getAlgorithm(algorithmId);
+		// First step is to check for any existing Jobs that match these parameters exactly. If so, then simply return
+		// that Job.
+		Job redundantJob = getExistingRedundantJob(sceneId, algorithm, computeMask);
+		if (redundantJob != null) {
+			// Add this Job to the Users Job table
+			// TODO - John help with adding job to users table
```

Comment:
Once i have the jobuser merged it should be easy to just use the jobuserdao to persist that info